src/base/rss/rss_autodownloader.cpp

```diff
@@ -245,6 +245,7 @@ void AutoDownloader::processJob(const QSharedPointer<ProcessingJob> &job)
         }
 
         rule.setLastMatch(job->articleDate);
+        rule.matched(job->articleTitle);
```

Comment:
Why do you have it outside matches() method? At least, this method has inappropriate name, since it's unclear what we're doing at this point.